wicket-core/src/main/java/org/apache/wicket/protocol/http/CsrfPreventionRequestCycleListener.java

```diff
@@ -451,6 +470,54 @@ protected void checkRequest(HttpServletRequest request, String sourceUri, IReque
 		}
 	}
 
+	/**
+     * Performs the check of the {@code Sec-Fetch-*} headers that are targeted at the
+     * {@code page}.
+     * @param request
+     *          the current container request
+     * @param sourceUri
+     *          the source URI
+     * @param page
+     *          the page that is the target of the request
+	 */
+	protected void checkRequestFetchMetadata(HttpServletRequest request, String sourceUri, IRequestablePage page)
+    {
+		//check if sourceUri exists before checking for whitelisted hosts,
+		// if not set sourceUri to no origin for logging purposes
+    	if (sourceUri == null || sourceUri.isEmpty())
```

Comment:
let's turn this guard into a single check

```suggestion
    	if (Strings.isEmpty(sourceUri))
```